experimental/segmenter/src/language.rs

```diff
@@ -12,115 +16,132 @@ pub enum Language {
     Unknown,
 }
 
-// TODO: Use data provider
-pub fn get_language(codepoint: u32) -> Language {
```

Comment:
Ok, so the main purpose of this PR is to resolve this TODO, right?

I think I'm not happy with requiring the full `ScriptWithExtensionsPropertyV1Marker` key for this. It is a fairly large key (>4 KB) replacing a very small lookup table. I'm not opposed to either making a smaller key, adding it to an existing payload, or just relying on a test to ensure this lookup is fresh. Note that we already are relying on hard-coded "th", "lo", "km", "my", and "ja" to load the data, so it's not unreasonable to have a code-driven lookup table in this isolated circumstance. If we do change this to be data-driven, we should also get rid of that hard-coded lookup.

Reply:
Yep. We could also just use `ScriptsV1`, but in its documentation it says that for continuation one should use `has_script` in case code points are in use by multiple scripts.